docker-compose-test.yml

```diff
@@ -117,10 +117,10 @@ services:
     container_name: gsconf4${COMPOSE_PROJECT_NAME}
     entrypoint: sleep infinity
     volumes:
-      - geoserver-data-dir:/geoserver_data/data
+      - geoserver-data-dir:${GEOSERVER_DATA_DIR}
```

Comment:
The image `geonode/geoserver_data:${VERSION}` will need to be adjusted if we intend it to be generic as per the env variable ${GEOSERVER_DATA_DIR}, Currently it also hard codes the path `/geoserver_data/data` so that's why I was reluctant to change it here unless we fix the upstream image. In my case, I am downloading the data directory as part of the GeoServer `entrypoint.sh` using `download.sh` and then mounting it to a specific path defined by `${GEOSERVER_DATA_DIR}`

Reply:
I will take a look and manage to see if the upstream image is customizable too